### PR TITLE
Adds libiconv dep

### PR DESCRIPTION
**Pull request by bkase**
Let me know if you want me to revert the auto-reformat, my editor wanted to reformat everything on save

### CODE REVIEW

File: nix/rust.nix [hunk 1]
```diff
@@ -96,76 +107,81 @@ in {
...
-      buildInputs = with final;
-        [ openssl ] ++ lib.optionals stdenv.isDarwin [
-          curl
-          darwin.apple_sdk.frameworks.Security
-        ];
+        buildInputs = with final;
+          [ openssl ] ++ lib.optionals stdenv.isDarwin [
+            curl
+            darwin.apple_sdk.frameworks.Security
+            libiconv
```
Comment (balsoft) [inline nix/rust.nix:140]: This change adds `libiconv` to inputs of `wasm-bindgen-cli`, which compiles fine as-is. The proper patch would be

```
diff --git a/nix/rust.nix b/nix/rust.nix
index 4a3b67dd30..4060cdcb33 100644
--- a/nix/rust.nix
+++ b/nix/rust.nix
@@ -72,6 +72,7 @@ in {
     ];
     sourceRoot = "source/lib/crypto/kimchi_bindings/stubs";
     nativeBuildInputs = [ final.ocamlPackages_mina.ocaml ];
+    buildInputs = with final; lib.optional stdenv.isDarwin libiconv;
     cargoLock = let fixupLockFile = path: builtins.readFile path;
     in {
       lockFileContents =
@@ -139,6 +140,7 @@ in {
     ];
     sourceRoot = "source/lib/crypto/kimchi_bindings/wasm";
     nativeBuildInputs = [ final.wasm-pack wasm-bindgen-cli ];
+    buildInputs = with final; lib.optional stdenv.isDarwin libiconv;
     cargoLock.lockFile = lock;
     cargoLock.outputHashes = narHashesFromCargoLock lock;
```

Comment (bkase) [reply nix/rust.nix:140]: This patch doesn't fix the error for me -- only my patch does

Comment (bkase) [reply nix/rust.nix:141]: ah the union of these two patches is good 👍

**Comment by bkase** [comment]
!ci-build-me

**Comment by balsoft** [comment]
I would prefer to have consistent formatting across the codebase, so please either switch to using `nixfmt`, disable autoformatting, or format everything using whatever formatter you have. It's easier for everyone this way.

**Comment by bkase** [comment]
fixed 👍 , don't know what weird formatter I had installed that wasn't nixfmt 🤷

**Comment by bkase** [comment]
!ci-build-me

**Review by balsoft — APPROVED** [review]

**Comment by balsoft** [comment]
Thanks!

**Pull request merged**